.github/workflows/ci-cd-workflow.yml

```
@@ -0,0 +1,248 @@
+name: pymagicc CI-CD
+on: push
+
+jobs:
+  linting-and-docs:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dev dependencies
+      run: |
+        pip install --upgrade pip
+        pip install -e .[dev]
+    # TODO: add `pylint pymagicc`
+    # TODO: add pydocstyle pymagicc
+    - name: Formatting and linters
+      run: |
+        black --check pymagicc tests setup.py --exclude pymagicc/_version.py
+        black-nb --check notebooks
+        isort --check-only --quiet --recursive pymagicc tests setup.py
+        bandit -c .bandit.yml -r pymagicc
+        flake8 pymagicc tests setup.py
+    - name: Build docs
+      # treat warnings as errors (-W)...
+      # ...but not when being nitpicky (-n)
+      run: |
+          sphinx-build -M html docs docs/build -qW
+          sphinx-build -M html docs docs/build -Eqn -b coverage
+          if [[ -s docs/build/html/python.txt ]]
+          then
+              echo
+              echo \"Error: Documentation missing:\"
+              echo
+              cat docs/build/html/python.txt
+              exit 1
+          fi
+
+  build:
+    needs: linting-and-docs
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # when we remove win, can add macOS back in
```

What's the actual limit of jobs?

---

It's unlikely that we catch 3.7/3.8 incompatibilities on Windows with this, so it might be enough to test different Python versions only on 1 platform maybe?

---

For open source, seems to be inifinte...

> GitHub Actions usage is free for public repositories.

https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions

---

```suggestion
        # when we remove wine, can add macOS back in
```

sorry that comment is wrong